### PR TITLE
Kat Fries: MVP done

### PR DESCRIPTION
**Pull request by cpoIT**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+// import db from './data/db.js'    // ES6 imports
+const server = express();
 
-// add your server code starting here
+
+// root of our site
+// req is a requestHandler
+server.get('/', (req, res) => {
+  // console.log(db.find());
+  res.send('<h2>API running on localhost:8000</h2><p>To see all posts: localhost:8000/api/posts</p></h2><p>To see a specific post by id (e.g., id = 1): localhost:8000/api/posts/1</p>')
```
Comment (kamry-bowman) [inline index.js]: Nice touch!

File: index.js [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+
+server.get('/api/posts/:id', (req, res) => {
+  const id = req.params.id;     // const { id } = req.params
+  console.log('params:', req.params); // params: { id: '1' }
+
+  db.findById(id)
+    .then(post => {
+      console.log(post) // logs id, title, contents, create_at, updated_at
+      // !(post.length) // fixed issue of empty array
+      !(post.includes(post[0])) // alternative to !(post.length)
```
Comment (kamry-bowman) [inline index.js]: I think the most straightforward way would be:
```
post.length
? res.status(200).json(post[0])
: res.status(404).json({ message: "The post with the specified ID does not exist."})
```

**Comment by cpoIT** [comment]
mvp day 2 done

File: index.js [hunk 3]
```diff
@@ -1,5 +1,108 @@
...
+
+// PUT/UPDATE
+server.put('/api/posts/:id', (req, res) => {
+  const { id } = req.params;
+  const changes = req.body;
+  console.log (changes)
+  db.update(id, changes)
+    .then(count => {
+      count 
+        ? res.status(200).json({ message: `updated post: { 'title': '${!changes.title ? 'title was not updated': changes.title}', 'contents': '${!changes.contents ? 'contents was not updated': changes.contents}' }`})
```
Comment (kamry-bowman) [inline index.js]: API calls for the full updated object to be sent back on success. Will need an additional api request to accomplish this

File: index.js [hunk 4]
```diff
@@ -1,5 +1,108 @@
...
+
+
+
+// DELETE
+server.delete('/api/posts/:id', (req, res) => {
+  db.remove(req.params.id)
+
+  .then(count => {
+    count
+    ? res.status(200).json(count) // don't return count return deleted post
```
Comment (kamry-bowman) [inline index.js]: API calls for full deleted post object to be returned, not just the count

File: index.js [hunk 5]
```diff
@@ -1,5 +1,108 @@
...
+server.get('/', (req, res) => {
+  // console.log(db.find());
+  res.send('<h2>API running on localhost:8000</h2><p>To see all posts: localhost:8000/api/posts</p></h2><p>To see a specific post by id (e.g., id = 1): localhost:8000/api/posts/1</p>')
+})
+
+// GET/READ ALL POSTS
+server.get('/api/posts', (req, res) => {
+  db.find()
+    .then(posts => {
+      res.status(200).json(posts)
```
Comment (kamry-bowman) [inline index.js]: I would suggest getting into the habit of returning from your .then blocks

File: index.js (hunk 3 above)
Comment (kamry-bowman) [inline index.js]: I would suggest returning the complete object, regardless of whether it was updated. Consider that in most cases the JSON object will be used to update something on the client side. It is much more useful to have the final version of the resource then a string describing the change

**Pull request closed without merging**